### PR TITLE
Cumulus

### PR DESCRIPTION
**Pull request by GopherJ**
- [x] use rococo-v1 dependencies, cumulus, substrate, polkadot-xcm and make it compiles
- [x] integrate our pallets
- [x] fix mock & tests
- [x] update README.md with parachain guide & parallel info
- [x] add stricter CI to enforce styles & tests & lint & wasm build
- [x] fix current CI
- [x] rabase latest changes of master
- [x] squash commits
- [x] add parallelfinance info(homepage etc)
- [ ] fix runtime cargo check & wasm check

### CODE REVIEW

File: node/Cargo.toml [hunk 1]
```diff
@@ -1,61 +1,78 @@
...
+polkadot-service      = { git = 'https://github.com/paritytech/polkadot.git', branch = 'rococo-v1' }
+polkadot-test-service = { git = 'https://github.com/paritytech/polkadot.git', branch = 'rococo-v1' }
+
+[build-dependencies]
+substrate-build-script-utils = { version = '3.0.0', git = 'https://github.com/paritytech/substrate.git', branch = 'rococo-v1' }
 
 [[bin]]
 name = 'parallel'
 
+[package]
```
Comment (kaichaosun) [inline node/Cargo.toml:63]: This section should be put at the beginning.

File: resources/para-200-genesis [hunk 2]
```diff
@@ -0,0 +1 @@
+0x0000000000000000000000000000000000000000000000000000000000000000006eceeb89f3ba476aff716ddb165a3aa307dcb0c5fa6348d067990e446b55f7f403170a2e7597b7b7e3d84c05391d139a62b157e78786d8c082f29dcf4c11131400
```
Comment (kaichaosun) [inline resources/para-200-genesis:1]: This file and wasm file should not in the git repo.

File: runtime/src/lib.rs [hunk 3]
```diff
@@ -92,38 +66,33 @@ pub mod opaque {
...
 
-/// This determines the average expected block time that we are targeting.
+/// This determines the average expected block time that we are targetting.
 /// Blocks will be produced at a minimum duration defined by `SLOT_DURATION`.
 /// `SLOT_DURATION` is picked up by `pallet_timestamp` which is in turn picked
 /// up by `pallet_aura` to implement `fn slot_duration()`.
 ///
 /// Change this to adjust the block time.
-pub const MILLISECS_PER_BLOCK: u64 = 6000;
+pub const MILLISECS_PER_BLOCK: u64 = 12000;
```
Comment (kaichaosun) [inline runtime/src/lib.rs:95]: Can you share more about this change?

File: runtime/src/lib.rs [hunk 4]
```diff
@@ -368,19 +266,15 @@ where
 
         let raw_payload = SignedPayload::new(call, extra)
             .map_err(|e| {
-                debug::warn!("SignedPayload error: {:?}", e);
+                log::warn!("SignedPayload error: {:?}", e);
```
Comment (kaichaosun) [inline runtime/src/lib.rs:269]: If error happens, can not make the log level to error?

File: runtime/src/lib.rs (hunk 3 above)
Comment (GopherJ) [reply runtime/src/lib.rs:95]: @kaichaosun from my understanding, rococo doesn't support 6000 block producing time. as gavin wood said during beijing meetings, now rococo only supports 12 seconds.

File: resources/para-200-genesis (hunk 2 above)
Comment (GopherJ) [reply resources/para-200-genesis:1]: while initializing a prarachain on polkadot.js we need this `genesis` data and the `wasm` file, I saw that some others repo are keeping them as well.

**Comment by yz89** [comment]
We can look at Acala Mandala TC, which is running in development mode.  They use the `alaca-dev' cli and its runtime to achieve this.

https://github.com/AcalaNetwork/Acala/tree/master/bin/acala-dev

**Comment by GopherJ** [comment]
> We can look at Acala Mandala TC, which is running in development mode. They use the `alaca-dev` cli and its runtime to achieve this.
> https://github.com/AcalaNetwork/Acala/tree/master/bin/acala-dev
> https://github.com/AcalaNetwork/Acala/blob/master/Makefile#L3
> 
> Maybe we can merge this pr firstly. Any suggestion about this? @kaichaosun

I tried this but it doesn't produce block yet

**Review by MrPai — COMMENTED** [review]
I can not compile successfully on a remote ubuntu system, neither the nightly nor stable toolchain. 
the error is "could not compile `polkadot-service`"

**Review by yz89 — APPROVED** [review]

**Review by kaichaosun — APPROVED** [review]
👍

**Pull request merged**